syntax = "proto3";

package user;

service UserService {
  rpc GetUserById (GetUserByIdRequest) returns (GetUserByIdResponse);
  rpc ValidateToken (ValidateTokenRequest) returns (ValidateTokenResponse);
}

message GetUserByIdRequest {
  int32 id = 1;
}

message GetUserByIdResponse {
  int32 id = 1;
  string username = 2;
  string role = 3;
}

message ValidateTokenRequest {
  string token = 1;
}

message ValidateTokenResponse {
  bool is_valid = 1;
  int32 user_id = 2;
  string role = 3;
}

